### ISSUE TITLE
azure-storage-cpp 5.0.0 doesn't compile on Visual Studio 2017

### ISSUE THREAD

**Issue by JiashuoLi** [Azure/azure-storage-cpp#199]
azure-storage-cpp 5.0.0 works fine with Visual Studio 2015, but when compiling with Visual Studio 2017, got error:

```
1>------ Build started: Project: ConsoleApplication1, Configuration: Debug Win32 ------
1>ConsoleApplication1.cpp
1>c:\users\xxx\source\repos\consoleapplication1\packages\cpprestsdk.v140.windesktop.msvcstl.dyn.rt-dyn.2.9.1\build\native\include\cpprest\asyncrt_utils.h(491): error C4596: 'system_type_to_datetime': illegal qualified name in member declaration
1>c:\users\xxx\source\repos\consoleapplication1\packages\cpprestsdk.v140.windesktop.msvcstl.dyn.rt-dyn.2.9.1\build\native\include\cpprest\streams.h(888): error C7510: 'int_type': use of dependent type name must be prefixed with 'typename'
1>c:\users\xxx\source\repos\consoleapplication1\packages\cpprestsdk.v140.windesktop.msvcstl.dyn.rt-dyn.2.9.1\build\native\include\cpprest\streams.h(1121): note: see reference to class template instantiation 'Concurrency::streams::basic_istream<_CharType>' being compiled
1>Done building project "ConsoleApplication1.vcxproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

This is a unknown issue of **cpprestsdk** 2.9.1 and is fixed in the latest release: https://github.com/Microsoft/cpprestsdk/issues/655

However, current **azure-storage-cpp** doesn't work with latest **cpprestsdk**: https://github.com/Azure/azure-storage-cpp/issues/134

**Comment by katmsft** [comment]
We are trying to push the 5.1.0 release that upgrades CPPRest SDK, however, we are currently blocked by [this issue](https://github.com/Microsoft/cpprestsdk/issues/804) of cpprestsdk. Please stay tuned for any updates.

**Comment by katmsft** [comment]
5.1.0 is now released and this issue is resolved.

**Comment by zowers** [comment]
shouldn't the issue be closed then?

**Comment by katmsft** [comment]
@zowers Yes, normally we keep the issue open for another day or two incase user wants to provide some feedback.

**Comment by JohnReeson** [comment]
@katmsft Version number not updated? I still see 5.0.0 using vcpkg search.

![1](https://user-images.githubusercontent.com/35160013/44620473-759d4700-a8c7-11e8-9159-8fb22d0eac4f.png)

**Comment by katmsft** [comment]
@JohnReeson Vcpkg picks up version update automatically, but it normally takes sometime. A new issue will be opened to track the version upgrade in Vcpkg.

**Comment by katmsft** [comment]
Now we have an opening issue tracking Vcpkg updates for azure-storage-cpp, so this issue will be closed.
